### PR TITLE
Modify symeig behavior when eigenvectors=False

### PR DESCRIPTION
**Pull request by vishwakftw**
This should be merged in tandem with https://github.com/pytorch/pytorch/pull/28338.

### CODE REVIEW

**Comment by vishwakftw** [comment]
cc: @ailzhang

**Comment by ailzhang** [comment]
Thanks @vishwakftw ! Did you run the xla test suite with both PRs btw? ;P

File: torch_xla/csrc/ops/symeig.cpp [hunk 1]
```diff
@@ -19,6 +19,10 @@ std::vector<xla::XlaOp> LowerSymEig(const xla::XlaOp& input, bool eigenvectors,
                           /*epsilon=*/1e-6);
   xla::XlaOp v = self_adj_eig_result.v;
   xla::XlaOp w = self_adj_eig_result.w;
+  if (!eigenvectors) {
+    v = xla::Zeros(input.builder(), xla::ShapeUtil::MakeShape(
+                                        input.shape().element_type(), {0}));
+  }
   if (!eigenvectors) {
```
Comment (ailzhang) [inline torch_xla/csrc/ops/symeig.cpp]: Should we remove this?

**Comment by vishwakftw** [comment]
Oh no, I don't have to the bandwidth to build XLA locally.

File: torch_xla/csrc/ops/symeig.cpp (hunk 1 above)
Comment (vishwakftw) [reply torch_xla/csrc/ops/symeig.cpp]: Oh, I missed this. My bad.

File: torch_xla/csrc/ops/symeig.cpp [hunk 2]
```diff
@@ -20,7 +20,9 @@ std::vector<xla::XlaOp> LowerSymEig(const xla::XlaOp& input, bool eigenvectors,
   xla::XlaOp v = self_adj_eig_result.v;
   xla::XlaOp w = self_adj_eig_result.w;
   if (!eigenvectors) {
-    v = xla::Zeros(input.builder(), XlaHelpers::ShapeOfXlaOp(v));
+    v = xla::Zeros(input.builder(),
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/symeig.cpp]: Just use `xla::Zero()`

Comment (vishwakftw) [reply torch_xla/csrc/ops/symeig.cpp]: Does it create an 0-numel `XlaOp` i.e., of shape `{0}`? I wasn't sure about it, so I went for the more verbose solution.

Comment (dlibenzi) [reply torch_xla/csrc/ops/symeig.cpp]: Yep, that's what it does:

https://github.com/tensorflow/tensorflow/blob/be20d163c9be8991feeb6bb39abe31ab55346110/tensorflow/compiler/xla/client/lib/constants.h#L121

File: test/cpp/test_aten_xla_tensor.cpp [hunk 3]
```diff
@@ -733,9 +733,13 @@ TEST_F(AtenXlaTensorTest, TestSymEig) {
           auto xla_b = torch::symeig(xla_a, eigenvectors, upper);
           AllClose(std::get<0>(b), std::get<0>(xla_b), /*rtol=*/3e-2,
                    /*atol=*/1e-2);
-          AllClose(std::get<1>(b).abs(), std::get<1>(xla_b).abs(),
-                   /*rtol=*/3e-2,
-                   /*atol=*/1e-2);
+          if (eigenvectors) {
```
Comment (dlibenzi) [inline test/cpp/test_aten_xla_tensor.cpp]: Hmm, why this?
Shouldn't pytorch return the same Zero we are returning now?
If not, what is the point of making the Zero change?

Comment (vishwakftw) [reply test/cpp/test_aten_xla_tensor.cpp]: PyTorch returns a 0-numel empty Tensor, which doesn’t have a value.

In ATen, we return `at::empty({0})`.

Comment (dlibenzi) [reply test/cpp/test_aten_xla_tensor.cpp]: Ouch!
Sorry to have misled you ... too many things going on.
Your previous code was correct ... brain *art on mine, sorry again.

Comment (vishwakftw) [reply test/cpp/test_aten_xla_tensor.cpp]: No problem, thanks! I've learned something new :-).

I'll revert the changes to `symeig.cpp`.

Comment (dlibenzi) [reply test/cpp/test_aten_xla_tensor.cpp]: Do we still need custom handling here?
Shouldn't comparing two tensors with a 0dim be returning equality?

Comment (vishwakftw) [reply test/cpp/test_aten_xla_tensor.cpp]: 0-dim tensors could be scalars or 0-numel tensors and comparisons become weird. For example:

`torch.tensor([]) == torch.tensor(0)` is `torch.tensor([], dtype=torch.bool`

**Comment by vishwakftw** [comment]
@dlibenzi @ailzhang the tests pass now, can this be merged?
